src/mail-service/index.ts

```diff
@@ -1,65 +1,53 @@
-import { formattedDate } from '../utils'
-import { getAllUnsentEmailData, updateNotificationSendStatus } from './db-query'
-import { EventType, eventMessages } from './email-message'
-import { sendMail } from './mail'
-import { EmailData, EmailPayload } from './types'
-
-export const filterDataForEmail = (emailData: EmailData[]) => {
-  const emailPayload: EmailPayload[] = []
-
-  emailData.forEach((data) => {
-    const dates = data.dates.split(',')
-    const event_ids = data.event_ids.split(',')
-    const notification_ids = data.notification_ids.split(',')
-    const actor_names = data.actor_names.split(',')
-    let body = ''
-    actor_names.forEach((actor: string, i: number) => {
-      body = `${body}<p>${actor} ${
-        eventMessages[parseInt(event_ids[i]) as EventType]
-      } ${formattedDate(new Date(dates[i]))}</p><br/>`
-    })
-    emailPayload.push({
-      user_id: data.user_id,
-      username: data.username,
-      email: data.email,
-      ids: notification_ids,
-      body,
+// eslint-disable-next-line import/no-internal-modules
+import { Connection } from 'mysql2/promise'
+import { Transporter } from 'nodemailer'
+
+import { getAllUnsentEmailData, updateNotificationSendStatus } from './queries'
+import { EmailPayload } from './utils'
+
+export async function notifyUsers(
+  connection: Connection,
+  transporter: Transporter,
+  senderEmailAddress: string
+): Promise<EmailPayload[]> {
+  const emailPayloads = await getAllUnsentEmailData(connection)
+
+  for (const payload of emailPayloads) {
```

Comment:
I'd prefer
```suggestion
  emailPayloads.forEach((payload) => {
```

Reply:
It would have to be a little more complicated as that (see [ec0890e](https://github.com/serlo/notification-mail-service/pull/13/commits/ec0890e9ce065a80dc367541e5b9b13c554a1010)). 
It's very common to use `.map` in our projects, but somehow it hasn't worked (see [fb26685](https://github.com/serlo/notification-mail-service/pull/13/commits/fb2668528c0e57c2a648258eb59a4de4c447dba5)). I suppose the connection closed  before the internal async calls were awaited.

Reply:
> It's very common to use .map in our projects, but somehow it hasn't worked (see [fb26685](https://github.com/serlo/notification-mail-service/pull/13/commits/fb2668528c0e57c2a648258eb59a4de4c447dba5)). I suppose the connection closed before the internal async calls were awaited.


and doing it with `.forEach` would lead to the same problem.